static/scripts/pay.ts

```diff
@@ -182,11 +182,14 @@ const fetchTreasury = async (): Promise<{ balance: number; allowance: number }>
   }
 };
 
-const toggleStatus = async (balance: number, allowance: number) => {
+const toggleStatus = async (balance: number, allowance: number, signer: JsonRpcSigner) => {
+  const tokenContract = new ethers.Contract(txData.permit.permitted.token, daiAbi, signer);
   const trBalance = document.querySelector(".tr-balance") as Element;
   const trAllowance = document.querySelector(".tr-allowance") as Element;
-  trBalance.textContent = balance ? `$${ethers.utils.formatUnits(balance, 18)}` : 'N/A';
-  trAllowance.textContent = balance ? `$${ethers.utils.formatUnits(allowance, 18)}` : 'N/A';
+  // get token decimals
+  const decimals = await tokenContract.decimals();
+  trBalance.textContent = balance ? `$${ethers.utils.formatUnits(balance, +decimals)}` : 'N/A';
+  trAllowance.textContent = balance ? `$${ethers.utils.formatUnits(allowance, +decimals)}` : 'N/A';
```

Comment:
I'm surprised that you need to coerce `decimals` into a number. 

Shouldn't `await tokenContract.decimals();` return a number?

Reply:
> I'm surprised that you need to coerce `decimals` into a number.
> 
> Shouldn't `await tokenContract.decimals();` return a number?

it is, i just thought the call might return a string

Reply:
I'd branch of to a new PR, but how do i disassociate this PR from the decimal issue

Reply:
You can try editing your original comment, and if that doesn't work, check the right side control panel on the GitHub UI
<img width="372" alt="Screenshot 2023-04-26 at 08 08 23" src="https://user-images.githubusercontent.com/4975670/234426539-4dfbf9f7-5917-4814-ad8f-8e802e6609c2.png">

Reply:
> You can try editing your original comment

Done, pull request 50 should close the new issue